great_expectations/rule_based_profiler/helpers/util.py

```diff
@@ -735,36 +773,103 @@ def compute_bootstrap_quantiles_point_estimate(
             np.random.PCG64(random_seed)
         )
         bootstraps = random_state.choice(
-            metric_values, size=(n_resamples, metric_values.size)
+            metric_values_converted, size=(n_resamples, metric_values_converted.size)
         )
     else:
         bootstraps = np.random.choice(
-            metric_values, size=(n_resamples, metric_values.size)
+            metric_values_converted, size=(n_resamples, metric_values_converted.size)
         )
 
-    lower_quantile_bias_corrected_point_estimate: np.float64 = _determine_quantile_bias_corrected_point_estimate(
+    lower_quantile_bias_corrected_point_estimate: Union[
+        np.float64, datetime.datetime
+    ] = _determine_quantile_bias_corrected_point_estimate(
         bootstraps=bootstraps,
         quantile_pct=lower_quantile_pct,
         quantile_statistic_interpolation_method=quantile_statistic_interpolation_method,
         quantile_bias_correction=quantile_bias_correction,
         quantile_bias_std_error_ratio_threshold=quantile_bias_std_error_ratio_threshold,
         sample_quantile=sample_lower_quantile,
     )
-    upper_quantile_bias_corrected_point_estimate: np.float64 = _determine_quantile_bias_corrected_point_estimate(
+    upper_quantile_bias_corrected_point_estimate: Union[
+        np.float64, datetime.datetime
+    ] = _determine_quantile_bias_corrected_point_estimate(
         bootstraps=bootstraps,
         quantile_pct=upper_quantile_pct,
         quantile_statistic_interpolation_method=quantile_statistic_interpolation_method,
         quantile_bias_correction=quantile_bias_correction,
         quantile_bias_std_error_ratio_threshold=quantile_bias_std_error_ratio_threshold,
         sample_quantile=sample_upper_quantile,
     )
+
+    if ndarray_is_datetime_type:
+        lower_quantile_bias_corrected_point_estimate = datetime.datetime.fromtimestamp(
+            lower_quantile_bias_corrected_point_estimate
+        )
+        upper_quantile_bias_corrected_point_estimate = datetime.datetime.fromtimestamp(
+            upper_quantile_bias_corrected_point_estimate
+        )
+
     return build_numeric_range_estimation_result(
         metric_values=metric_values,
         min_value=lower_quantile_bias_corrected_point_estimate,
         max_value=upper_quantile_bias_corrected_point_estimate,
     )
 
 
+def build_numeric_range_estimation_result(
+    metric_values: np.ndarray,
```

Comment:
Again, why not use `MetricValues` here since it is an alias for `np.ndarray`?

Reply:
@NathanFarmer Just convention (per previous answer).